core/src/main/java/c/org/rajawali3d/control/RenderSurfaceView.java

```diff
@@ -0,0 +1,37 @@
+package c.org.rajawali3d.control;
+
+import c.org.rajawali3d.surface.SurfaceView;
+
+/**
+ * Internal extensions of {@link SurfaceView} required by a {@link RenderControl} implementation. The goal is to
+ * define a minimal interface that can reasonably be expected to be supportable by current (GLES) and future
+ * (Vulkan?) graphics systems.
+ *
+ * @author Randy Picolet
+ */
+public interface RenderSurfaceView extends SurfaceView {
+
+    /**
+     * Sets whether frames are rendered only on request (and when the surface is first created), or continuously
+     *
+     * @param onRequest {@code boolean} true to render on request, false to render continuously
+     */
+    void setRenderFramesOnRequest(boolean onRequest);
+
+    /**
+     * Request rendering of the next complete frame for the SurfaceView.
+     */
+    void requestFrameRender();
+
+    /**
+     *
+     * @param runnable
+     */
+    void queueEvent(Runnable runnable);
```

Comment:
Can you please elaborate on the reasoning for both `queueEvent` and `post`?

Reply:
queueEvent() posts to the render thread, post() to the main/UI thread

Reply:
I would prefer method names which reflected that.

Reply:
[Not sure if my edit was before or after your comment, so moving foreard I will stop editing except for spelling and such, not semantics or content]

Since the methods are already present in both GlesSurfaceView and GlesSurfaceTextureView, I just used them as-is so those classes did not have to be changed or introduce another delegation, but it makes sense looking forward to more surface view impls e.g. for Vulkan.  I will change to `queueToRenderThread()` and `queueToMainThread()`.  Good call.

Reply:
Sounds good. Also, with our rather loose discussion about possibly extracting the render thread class, we basically will be keeping our own version of both TextureView and SurfaceView, so we can ultimately call them whatever we want.